client/logmon/logmon.go

```diff
@@ -204,10 +204,7 @@ func newLogRotatorWrapper(path string, logger hclog.Logger, rotator io.WriteClos
 	var openFn func() (io.ReadCloser, error)
 	var err error
 
-	//FIXME Revert #5990 and check os.IsNotExist once Go >= 1.12 is the
```

Comment:
I decided not to revert https://github.com/hashicorp/nomad/pull/5990 completely as it contained some testing changes and some refactoring that we may want to keep.